src/components/plan/PlanForm.svelte

```diff
@@ -133,12 +141,92 @@
       effects.updatePlan(plan, { name: $planNameField.value }, user);
     }
   }
+
+  async function exportPlan() {
+    if (plan) {
+      if (planExportAbortController) {
+        planExportAbortController.abort();
+      }
+
+      planExportAbortController = new AbortController();
+
+      let qualifiedActivityDirectives: ActivityDirective[] = [];
+      planExportProgress = 0;
+
+      let totalProgress = 0;
+      const numOfDirectives = Object.values(activityDirectivesMap).length;
+
+      qualifiedActivityDirectives = await Promise.all(
+        Object.values(activityDirectivesMap).map(async activityDirective => {
+          if (plan) {
+            const effectiveArguments = await effects.getEffectiveActivityArguments(
+              plan?.model_id,
+              activityDirective.type,
+              activityDirective.arguments,
+              user,
+              planExportAbortController?.signal,
+            );
+
+            totalProgress++;
+            planExportProgress = (totalProgress / numOfDirectives) * 100;
+
+            return {
+              ...activityDirective,
+              arguments: effectiveArguments?.arguments ?? activityDirective.arguments,
+            };
+          }
+
+          totalProgress++;
+          planExportProgress = (totalProgress / numOfDirectives) * 100;
+
+          return activityDirective;
+        }),
+      );
+
+      if (!planExportAbortController.signal.aborted) {
```

Comment:
Seeing this error if I throttle my network and click rapidly on the download button
<img width="781" alt="image" src="https://github.com/NASA-AMMOS/aerie-ui/assets/4419607/b5d093de-b0c5-4b03-af51-42ec7b2c2be1">

Reply:
And the button is stuck in the downloading state and I can't initiate more downloads

Reply:
@AaronPlave good catch! I think this should be ready for another look. 🤞🏻